### ISSUE TITLE
doesn't work in shell-mode?

### ISSUE THREAD

**Issue by snarfed** [rakete/too-long-lines-mode#1]
`too-long-lines-mode` is great! thank you for writing it. i noticed it doesn't work in `shell-mode` though. it enables fine, but `too-long-lines-hide` doesn't do anything. any chance you might be able to fix that? thanks in advance!

**Comment by rakete** [comment]
I changed the behaviour of `too-long-lines-mode` and it should now work in `shell-mode` as well.

For performance reasons I don't put `too-long-lines-hide` into `after-change-functions` anymore. Instead I now create a timer that periodically calls `too-long-lines-hide` in buffers that are in a major-mode from the variable `too-long-lines-special-buffer-modes`.

**Comment by snarfed** [comment]
awesome, thank you for working on this! `too-long-lines-hide` works in shell buffers for me now, and the timer does too.

it still doesn't quite handle my use case, though. sometimes i'll have commands or other output printing a single long line in a shell buffer for many second or tens of seconds, which gets slower and slower as it gets longer, due both to font-locking and to other emacs internals.

i'd love `too-long-lines-mode` to hide those lines even while they're still printing, instead of only after the buffer goes idle. and then after it hides a line, new characters appended to that line would ideally be hidden immediately too.

i know this may not be easy. thank you again for all your help so far!

**Comment by rakete** [comment]
Thanks for actually testing this and giving me feedback!

I changed the code again and fixed the other issue with the timer not being cancelled you reported.

But even better I think I actually solved the hiding of lines in shell-mode buffers much better now. I changed the code so that `too-long-lines-hide` can be used in `after-change-functions` again, and because I previously changed the code so that an existing hiding overlay in a line which expanded beyond that overlay would get deleted and recreated again, it now works much better in `shell-mode`.

So instead of using a timer I add `too-long-lines-hide-after-change-in-special-buffer` to `after-change-functions`, which calls `too-long-lines-hide` only in buffers with modes defined in `too-long-lines-special-buffer-modes`. Dealing with overlays is notoriously slow in emacs, and `after-change-functions` get called a lot while editing, which means there is a lot of potential to make emacs laggy when adding something that deals with overlays to `after-change-functions`.
